weather_mv/loader_pipeline/ee_util.py

```diff
@@ -0,0 +1,457 @@
+        # with EE from here: https://signup.earthengine.google.com/#!/service_accounts
+        # See https://developers.google.com/earth-engine/guides/service_account#register-the-
+        # service-account-to-use-earth-engine
+        # for more detail.
+        creds = compute_engine.Credentials()
+    else:
+        creds, _ = default()
+    request = requests.Request()
+    creds.refresh(request)
+    return creds
+
+
+def ee_initialize(use_personal_account=False,
+                  enforce_high_volume=False,
+                  allow_old_versions=False):
+    """Initializes ee with the high volume API when using a compute engine VM.
+
+    Once earth engine is initialized, repeated calls will still verify the high
+    volume api and old version constrains, but will then no-op.
+
+    Raises:
+        RuntimeError: Earth Engine did not initialize.
+    """
+    creds = get_creds(use_personal_account)
+
+    # Only the compute engine service service account can access the high volume api.
+    # The high volume api is only available with ee version 0.1.263 or later.
+
+    up_to_date = ee.__version__ >= '0.1.263'
+    on_compute_engine = is_compute_engine()
+
+    # Try to use the high volume api
+    if up_to_date and on_compute_engine:
+        ee.Initialize(creds, opt_url='https://earthengine-highvolume.googleapis.com')
+
+    # If the high volume api is required, find out why
+    elif enforce_high_volume and not on_compute_engine:
+        raise RuntimeError(
+            'Must run on a compute engine VM to use the high volume earth engine api.'
+        )
+    elif enforce_high_volume and not up_to_date:
+        raise RuntimeError(
+            'Earth engine api must be updated to version ≥0.1.263 to use the high volume api'
+        )
+
+    # Ensure the api version is acceptable
+    elif not up_to_date and not allow_old_versions:
+        raise RuntimeError(
+            'Earth engine api is out of date. Update to version ≥0.1.263.')
+
+    # Try using the service account
+    elif on_compute_engine:
+        ee.Initialize(creds)
+
+    # Try using an personal authentication
+    else:
+        ee.Initialize()
+
+
+@dataclasses.dataclass
+class GribData(object):
+    """A class for holding the output of the GetChannelsDoFn."""
+    name: str  # The EE-safe name of the channel.
+    start_time: float  # In floating point seconds since epoch.
+    end_time: float  # In floating point seconds since epoch.
+    dtype: str  # data type of the channel data
+    projection: str  # The gdal string defining the projection.
+    transform: t.Tuple[float, float, float, float, float, float]  # From gdal.
+    metadata: t.Dict[str, t.Union[str, float, int]]
+    data: t.List[xr.Dataset]  # The list of datasets
+
+
+@dataclasses.dataclass
+class ToEarthEngine(ToDataSink):
+    """Loads weather data into Google Earth Engine.
+
+    TODO(alxr): Document...
```

Comment:
Please omit this... or, file a Github issue, and associate the TODO with the issue, e.g. 
```
TODO(#180): Document...
```

Reply:
Or, better yet: please add the class documentation :)